lib/_stream_readable.js

```diff
@@ -526,6 +537,14 @@ Readable.prototype.pipe = function(dest, pipeOpts) {
     }
   }
 
+  // if src has an error and error forwarding is turned on
+  // cause the error to be emited in dest and do suppress the
+  // throwing behavior
+  function onsrcerror(er) {
+    debug('onsrcerror', er);
+    dest.emit('error', er);
```

Comment:
Do we care if this creates an infinite loop? `a.pipe(b, {forwardErrors: true}).pipe(a, {forwardErrors: true})`